src/terrain/terrain.hpp

```diff
@@ -145,8 +145,6 @@ class terrain_type
 	//these descriptions are shown for the terrain in the mouse over
 	//depending on the owner or the village
 	const t_string& income_description() const { return income_description_; }
-	const t_string& income_description_ally() const { return income_description_ally_; }
-	const t_string& income_description_enemy() const { return income_description_enemy_; }
```

Comment:
At the first moment, I tried to create a method here, `income_description_other`.
But since it could not be a `const`, and `map.get_terrain_info(terrain)` returns a const, I needed to move this code completely to `reports.cpp`.

Reply:
This change eliminates two (maybe three) keys from the terrain WML, so it feels like a regression. I think the idea is good, but we probably need to work out a different presentation.

In particular, it appears we lose the ability for an income tile to be labelled as anything other than a village. Prior to this change, it seems like we'd have the ability to create a tile that shows as "Bank", "Allied Bank", "Enemy Bank", etc. After this change, such a tile would now be referred to as a village if I'm not mistaken.

Reply:
We still can translate this by `$village_owner's village`
https://github.com/wesnoth/wesnoth/blob/9e7459600f228433c8abd8eed229f78e1a2290ec/src/reports.cpp#L1578

I'm not sure how WML works on it, but if WML can change the same strings as translations do, we'll be able to update it there as well.

But yeah, it'll be a regression. It's a breaking change.